File: src/main/java/com/finances/repository/SavingsRepository.java
package com.finances.repository;

import com.finances.entity.Savings;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.jpa.repository.support.JpaRepositoryImplementation;
import org.springframework.stereotype.Repository;

import javax.transaction.Transactional;
import java.math.BigDecimal;

@Repository
@Transactional
public interface SavingsRepository extends JpaRepositoryImplementation<Savings, Integer> {

    @Query("SELECT SUM(s.value) FROM Savings s")
    BigDecimal getSumSavings();
}

File: src/main/kotlin/com/finances/repository/SavingsRepository.kt
package com.finances.repository

import com.finances.entity.Savings
import org.springframework.data.jpa.repository.Query
import org.springframework.data.jpa.repository.support.JpaRepositoryImplementation
import org.springframework.stereotype.Repository
import java.math.BigDecimal
import javax.transaction.Transactional

@Repository
@Transactional
interface SavingsRepository : JpaRepositoryImplementation<Savings, Int> {

    @Query("SELECT SUM(s.value) FROM Savings s")
    fun sumSavings(): BigDecimal
}
